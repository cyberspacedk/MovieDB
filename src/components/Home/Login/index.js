import LoginForm from './container';

export default LoginForm;
