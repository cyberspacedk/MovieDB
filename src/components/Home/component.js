import React from 'react';
import PropTypes from 'prop-types';
import Dashboard from './Dashboard';
import Login from './Login';

const Home = ({ isAuthentificated }) => (
  <>{!isAuthentificated ? <Login /> : <Dashboard />}</>
);

Home.propTypes = {
  isAuthentificated: PropTypes.bool,
};
export default Home;
