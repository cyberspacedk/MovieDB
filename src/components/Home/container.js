import { connect } from 'react-redux';
import { isAuthentificated } from '../../store/authentifiction/selectors';
import Home from './component';

export const mstp = state => ({
  isAuthentificated: isAuthentificated(state),
});

export default connect(mstp)(Home);
