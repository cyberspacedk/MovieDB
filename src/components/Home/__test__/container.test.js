import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Home from '../container';

describe('Component: Home', () => {
  const store = configureStore()({
    user: {
      sessionId: 'jchdi3bmc99aj',
    },
  });
  const container = shallow(<Home store={store} />);

  it('Match its snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
