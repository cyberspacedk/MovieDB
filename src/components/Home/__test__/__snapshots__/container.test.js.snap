// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Home Match its snapshot 1`] = `
<ContextProvider
  value={null}
>
  <Home
    dispatch={[Function]}
    isAuthentificated={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
